src/services/observables/tokens.ts

```diff
@@ -399,59 +407,73 @@ export const poolTokens$ = combineLatest([
       converter: string;
     } | null>(
       partialPoolTokens.map(async (poolToken) => {
-        const pool = pools.find(
+        let pool: APIPool | Pool;
```

Comment:
this code, from here till line 426 - should go in a helper function that returns the pool or null

Reply:
done